src/main/java/net/lldp/checksims/token/TokenList.java

```diff
@@ -111,15 +112,20 @@ public String join(boolean onlyValid) {
     }
 
     /**
-     * Peforms a shallow copy of a TokenList, returning an immutable version of the initial list.
+     * Peforms a deep copy of a TokenList, returning an immutable version of the initial list with immutable tokens.
      *
      * @param cloneFrom List to copy
      * @return Immutable copy of cloneFrom
      */
-    public static TokenList immutableCopy(TokenList cloneFrom) {
+    public static TokenList immutableCopy(final TokenList cloneFrom) {
         checkNotNull(cloneFrom);
 
-        return new TokenList(cloneFrom.type, ImmutableList.copyOf(cloneFrom));
+        final List<ImmutableToken> tmp = new LinkedList<>();
```

Comment:
I want to say this should be an ArrayList... But since we aren't returning it, doesn't really matter.